lib/Helper/HTMLParser/HttpMicrodataParser.php

```diff
@@ -0,0 +1,265 @@
+	 * @param \DOMNode $recipeNode The recipe to search within
+	 * @return boolean true, if a property was found
+	 */
+	private function parseIngredients(\DOMNode $recipeNode) {
+		return $this->searchMultipleProperties($recipeNode,
+			['recipeIngredient', 'ingredients'],
+			['content'],
+			'recipeIngredient');
+	}
+	
+	/**
+	 * Search for instructions in the microdata of a recipe
+	 * @param \DOMNode $recipeNode The recipe to search within
+	 * @return boolean true, if a property was found
+	 */
+	private function parseInstructions(\DOMNode $recipeNode) {
+		return $this->searchMultipleProperties($recipeNode,
+			['recipeInstructions', 'instructions', 'steps', 'guide'],
+			['content'],
+			'recipeInstructions');
+	}
+	
+	/**
+	 * Search for microdata properties under various names in the recipe and save as a list
+	 *
+	 * If a property can be named differntly and found under differnt property names within
+	 * the DOM tree, this method looks for all these options.
+	 * It is similar to the searchSimpleProperties() method but allows to search for different
+	 * names within the DOM tree. This can be useful when a property is superseeded and the code
+	 * should still be backward compatible.
+	 *
+	 * @param \DOMNode $recipeNode The node of the recipe to look for properties in
+	 * @param array $properties The properties to look for one-by-one
+	 * @param array $attributes The attributes that will contain the data
+	 * @param string $dst The name of the property list in the internal structure
+	 * @return boolean true, if the property was found
+	 */
+	private function searchMultipleProperties(\DOMNode $recipeNode, array $properties,
+		array $attributes, string $dst) {
+		foreach ($properties as $prop) {
+			$entries = $this->searchChildEntries($recipeNode, $prop);
+			
+			try {
+				$arrayObtained = $this->extractAttribute($entries, $attributes);
+				
+				if (count($arrayObtained) > 0) {
+					$this->recipe[$dst] = $arrayObtained;
+					return true;
+				}
+			} catch (AttributeNotFoundException $ex) {
+				// Test with the next property name
+				continue;
+			}
+		}
+		
+		return false;
+	}
+	
+	/**
+	 * Search for microdata properties of the recipe.
+	 *
+	 * Within the recipe node search for DOM entries that represent a property and assign the
+	 * internal structure to their value.
+	 *
+	 * @param \DOMNode $recipeNode The node of the recipe under which to look for properties
+	 * @param string $property The property name to look for
+	 * @return bool true, if the property was found
+	 */
+	private function seachSimpleProperties(\DOMNode $recipeNode, string $property): bool {
+		$entries = $this->searchChildEntries($recipeNode, $property);
+		
+		try {
+			$values = $this->extractAttribute($entries, ['content']);
+			
+			if (count($values) === 1) {
+				$values = $values[0];
+			}
+		} catch (AttributeNotFoundException $ex) {
+			return false;
+		}
+		
+		$this->recipe[$property] = $values;
+		return true;
```

Comment:
Can this be replaced by a call to `searchMultipleProperties`?

Reply:
Should be done.